### DISCUSSION TITLE
Interested in implementing feature, looking for initial feedback

### DISCUSSION THREAD

**Discussion by CJEnright** [vercel/turborepo#2503] [Help] [answered]
I'd like to implement #864, but wanted to do an initial feasibility/maintainer check.

The simplest solution I could think of would to be make `prune`'s `--scope` option a comma separated list, then split the [scope string here](https://github.com/vercel/turbo/blob/main/cli/internal/prune/prune.go#L143) and grab ancestors for each scope then deduplicate.

Alternatively, I could try to use [`scope.Opts`](https://github.com/vercel/turbo/blob/main/cli/internal/scope/scope.go#L47) like `run` uses in order to make things more flexible. This may also be preferred because it [looks like the `--scope` flag is being deprecated](https://turbo.build/repo/docs/reference/command-line-reference#--scope).

Would one of these be preferred? Is this even something that the maintainers are interested in including? I also know there's an effort to migrate the go code to rust, so if it'd be best to wait until after that's complete please let me know!

**Comment by nathanhammond** [comment, marked as answer]
So, there is a much easier method to do this on the input side, similar to the way we do `--filter`:
https://github.com/vercel/turbo/blob/60f143e58735fa9b846035763ff9653ca4d957fd/cli/internal/scope/scope.go#L71

You'd make that change here:
https://github.com/vercel/turbo/blob/60f143e58735fa9b846035763ff9653ca4d957fd/cli/internal/prune/prune.go#L28

This allows specifying multiple `--scope` arguments.

Once you've received the input it's a _little_ bit interesting in how that gets applied to `prune`. @chris-olszewski will be the person who can give more details if you get stuck!

**Reply by CJEnright** [reply]
Thanks for the tips! This ended up not being too bad to get sorted. I've opened a PR here with what I ended up doing: https://github.com/vercel/turbo/pull/2551
